### DISCUSSION TITLE
Path categories for middleware

### DISCUSSION THREAD

**Discussion by tinleym** [vercel/next.js#42880] [Ideas]
### Describe the feature you'd like to request

It would be nice to get path categories like "PAGE" or "PUBLIC" for middleware, to more clearly and robustly distinguish between different types of requests.

### Describe the solution you'd like

I think a parameter on req.nextUrl would do it.

### Describe alternatives you've considered

The existing regex method gets the job done, but it seems like it would be clearer and easier to narrow by request type explicitly rather than have to manage a string matching rule.

**Comment by balazsorban44** [comment]
Hi, this was the case before, but given that Middleware runs before the routing logic, it cannot know for certain if a request is for a page or not.

See also: https://github.com/vercel/next.js/issues/34521

**Reply by tinleym** [reply]
Thank you!
